packages/x-components/src/wiring/wiring.types.ts

```diff
@@ -34,7 +34,7 @@ export type Wire<PayloadType> = (
  *
  * @public
  */
-export interface WireMetadata {
+export interface WireMetadata<PayloadType = any> {
```

Comment:
Thanks to this change I've just realised that the wiring types are wrong when you use a `map` function. I've tried to fix it, but its too complex so I've created [a task](https://searchbroker.atlassian.net/browse/EX-5192) to properly check it. My proposal would be to just use the `mapWire` operator.